core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/UserCoreUtil.java

```diff
@@ -582,6 +582,9 @@ public static String setDomainToUpperCase(String name) {
      */
     public static String addDomainToName(String name, String domainName) {
 
+        if (StringUtils.isEmpty(name)) {
+            return null;
```

Comment:
Why are we returning null? Can't we just return the name? Retuning null may introduce NPE for the existing logic?